signer/src/transaction_coordinator.rs

```diff
@@ -2753,14 +2758,62 @@ mod tests {
         more_asserts::assert_gt!(start.elapsed(), delay + baseline_elapsed);
     }
 
+    /// Check that we skip processing bitcoin blocks if the chain tip in
+    /// the state doesn't match the block hash passed in.
+    #[tokio::test]
+    async fn should_skip_processing_bitcoin_blocks_if_chain_tip_has_changed() {
+        let mut rng = testing::get_rng();
+        let ctx = TestContext::builder()
+            .with_in_memory_storage()
+            .with_mocked_clients()
+            .modify_settings(|settings| {
+                settings.signer.bootstrap_signatures_required = 1;
+                settings.signer.bootstrap_signing_set =
+                    std::iter::once(settings.signer.public_key()).collect();
+            })
+            .build();
+
+        let network = WanNetwork::default();
+        let net = network.connect(&ctx);
+
+        let mut ev = TxCoordinatorEventLoop {
+            network: net.spawn(),
+            context: ctx.clone(),
+            context_window: 10000,
+            private_key: ctx.config().signer.private_key,
+            signing_round_max_duration: Duration::from_secs(10),
+            bitcoin_presign_request_max_duration: Duration::from_secs(10),
+            threshold: ctx.config().signer.bootstrap_signatures_required,
+            dkg_max_duration: Duration::from_secs(10),
+            is_epoch3: true,
+        };
+
+        let chain_tip1: BitcoinBlockRef = fake::Faker.fake_with_rng(&mut rng);
+        let chain_tip2: BitcoinBlockRef = fake::Faker.fake_with_rng(&mut rng);
+
+        // There is only one signer in the signer set, us, so we should
+        // always be coordinator.
+        assert!(ev.is_coordinator(&chain_tip1.block_hash));
+        assert!(ev.is_coordinator(&chain_tip2.block_hash));
+
+        // We should bail if the chain tip is not set.
+        let error = ev.process_new_blocks(chain_tip2).await.unwrap_err();
+        assert_matches::assert_matches!(error, Error::NoChainTip);
+
+        // Now the chain tip in the state is set but it does not match the
+        // chain tip passed in, so we should bail early and return Ok(())
+        ctx.state().set_bitcoin_chain_tip(chain_tip1);
+        ev.process_new_blocks(chain_tip2).await.unwrap();
```

Comment:
Can you also add the match err if we try to process `chain_tip2` after setting that in the state (or process `chain_tip1`)? Otherwise is not obvious if/what should fail, and isn't failing because of the early return.

Reply:
Nice, just the comment is maybe missing some word or a bad copypaste?